src/tap/src/tapcell.tcl

```diff
@@ -1108,13 +1108,14 @@ proc remove_cells {prefix} {
 }
 
 proc check_symmetry {master ori} {
-  set symmetry_x [$master getSymmetryX]
-  set symmetry_y [$master getSymmetryY]
-
-  set symmetry_ok 1
-  if {(!$symmetry_x && $ori == "MX") || \
-      (!$symmetry_y && $ori == "R0")} {
-    set symmetry_ok 0
+  set symmetry_x_only [expr [$master getSymmetryX] && ![$master getSymmetryY]]
+  set symmetry_y_only [expr [$master getSymmetryY] && ![$master getSymmetryX]]
+
+  set symmetry_ok 0
+  if {($symmetry_x_only && $ori == "MX") || \
+      ($symmetry_y_only && $ori == "R0") || \
+      (!$symmetry_x_only && !$symmetry_y_only)} {
```

Comment:
With x_only you can have MX or R0 and with y_only you can have MY or R0.  

You also need to cover the no symmetry case (neither X nor Y for which only R0 is allowed).

I'm thinking more of (pseudo-code)

switch orientation
case R0 : return true
case MX : return sym_x
case MY:  return sym_y
case R180: return sym_x & sym_y
default: return false
}